### ISSUE TITLE
JSpinner Horizontal GroupLayout Issue

### ISSUE THREAD

**Issue by RobertBColton** [mgarin/weblaf#386]
In our preferences frame we are using a GroupLayout to create backup preferences.

```java
		JPanel backupsPanel = new JPanel();
		backupsPanel.setBorder(BorderFactory.createTitledBorder("Backups"));

		JLabel maxCopiesLabel = new JLabel("Copies:");
		JSpinner maxCopiesField = new JSpinner(new SpinnerNumberModel(Prefs.backupCopies, 0, 5, 1));
		JLabel hoursLabel = new JLabel("Hours:");
		JSpinner hoursField = new JSpinner(new SpinnerNumberModel(Prefs.backupHours, 0, 12, 1));
		JLabel minutesLabel = new JLabel("Minutes:");
		JSpinner minutesField = new JSpinner(new SpinnerNumberModel(Prefs.backupMinutes, 0, 59, 1));

		GroupLayout backupsLayout = new GroupLayout(backupsPanel);
		backupsLayout.setAutoCreateGaps(true);
		backupsLayout.setAutoCreateContainerGaps(true);
		backupsPanel.setLayout(backupsLayout);

		backupsLayout.setHorizontalGroup(backupsLayout.createSequentialGroup()
		/**/.addGroup(backupsLayout.createParallelGroup(Alignment.TRAILING)
		/*	*/.addComponent(maxCopiesLabel)
		/*	*/.addComponent(hoursLabel)
		/*	*/.addComponent(minutesLabel))
		/**/.addGroup(backupsLayout.createParallelGroup()
		/*	*/.addComponent(maxCopiesField)
		/*	*/.addComponent(hoursField)
		/*	*/.addComponent(minutesField)));

		backupsLayout.setVerticalGroup(backupsLayout.createSequentialGroup()
		/**/.addGroup(backupsLayout.createParallelGroup(Alignment.BASELINE)
		/*	*/.addComponent(maxCopiesLabel)
		/*	*/.addComponent(maxCopiesField))
		/**/.addGroup(backupsLayout.createParallelGroup(Alignment.BASELINE)
		/*	*/.addComponent(hoursLabel)
		/*	*/.addComponent(hoursField))
		/**/.addGroup(backupsLayout.createParallelGroup(Alignment.BASELINE)
		/*	*/.addComponent(minutesLabel)
		/*	*/.addComponent(minutesField)));

		JPanel pPanel = new JPanel();
		GroupLayout pLayout = new GroupLayout(pPanel);
		pLayout.setAutoCreateGaps(true);
		pLayout.setAutoCreateContainerGaps(true);
		pPanel.setLayout(pLayout);

		pLayout.setHorizontalGroup(pLayout.createParallelGroup().addComponent(backupsPanel,DEFAULT_SIZE,DEFAULT_SIZE,PREFERRED_SIZE));
		pLayout.setVerticalGroup(pLayout.createSequentialGroup().addComponent(backupsPanel));

		return pPanel;
```

There seems to be a slight inconsistency between the Web L&F and other look and feels. You can see in the following screenshot the spinbox used for copies is not the same width as the other spinboxes. The look and feels in this picture are native, Nimbus, Swing cross-platform, and Web L&F from left to right.
![JSpinner Horizontal GroupLayout](https://cloud.githubusercontent.com/assets/3212801/15267451/9a1972ca-1990-11e6-867d-ee9c1db9852d.png)

This issue does not occur in 1.28 but does occur in the latest 1.29.

**Comment by mgarin** [comment]
Thanks for the example!
I will investigate why that happens.

**Comment by mgarin** [comment]
This is one of the old issues which have been around since custom preferred size have been added to most components inside the UI. It seems to interfere with some `GroupLayout` logic and cause different behavior. 

I have pushed a small fix for `JSpinner` component, it will be available in snapshot build tomorrow.

Also new snapshot builds are now available for other changes I have made before this one:
https://oss.sonatype.org/content/repositories/snapshots/com/weblookandfeel/
http://weblookandfeel.com/downloads/styling/
